fenix/app/src/androidTest/java/org/mozilla/fenix/ui/NimbusMessageHomescreenTest.kt

```diff
@@ -0,0 +1,119 @@
+package org.mozilla.fenix.ui
+
+import android.content.Intent
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.uiautomator.UiDevice
+import okhttp3.mockwebserver.MockWebServer
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.mozilla.experiments.nimbus.Res
+import org.mozilla.fenix.components.appstate.AppAction
+import org.mozilla.fenix.ext.components
+import org.mozilla.fenix.helpers.AndroidAssetDispatcher
+import org.mozilla.fenix.helpers.HomeActivityIntentTestRule
+import org.mozilla.fenix.helpers.RetryTestRule
+import org.mozilla.fenix.helpers.TestHelper.appContext
+import org.mozilla.fenix.nimbus.FxNimbus
+import org.mozilla.fenix.nimbus.HomeScreenSection
+import org.mozilla.fenix.nimbus.Homescreen
+import org.mozilla.fenix.nimbus.MessageData
+import org.mozilla.fenix.nimbus.MessageSurfaceId
+import org.mozilla.fenix.nimbus.Messaging
+import org.mozilla.fenix.nimbus.StyleData
+import org.mozilla.fenix.ui.robots.homeScreen
+
+class NimbusMessageHomescreenTest {
+    private lateinit var mDevice: UiDevice
+    private lateinit var mockWebServer: MockWebServer
+
+    private var messageButtonLabel = "CLICK ME"
+    private var messageText = "Some Nimbus Messaging text"
+    private var messageTitle = "A Nimbus title"
+
+    @get:Rule
+    val homeActivityTestRule = HomeActivityIntentTestRule.withDefaultSettingsOverrides(
+        skipOnboarding = true,
+    ).withIntent(
+        Intent().apply {
+            action = Intent.ACTION_VIEW
+        },
+    )
+
+    @Rule
+    @JvmField
+    val retryTestRule = RetryTestRule(3)
+
+    @Before
+    fun setUp() {
+        // Set up nimbus message
+        FxNimbus.features.messaging.withInitializer {
+            // FML generated objects.
+            Messaging(
+                messages = mapOf(
+                    "test-message" to MessageData(
+                        action = Res.string("TEST ACTION"),
+                        style = "URGENT",
```

Comment:
Should this be `TEST STYLE`?

Reply:
Yes, good catch, I think it was a carry over from debugging